kbs/src/api/src/attestation/mod.rs

```diff
@@ -32,6 +32,9 @@ pub trait Attest: Send + Sync {
     /// Verify Attestation Evidence
     /// Return Attestation Results Token
     async fn verify(&self, tee: Tee, nonce: &str, attestation: &str) -> Result<String>;
+
+    /// generate the challenge payload to pass to attester based on Tee and nonce
+    async fn generate_challenge(&self, tee: Tee, nonce: &str) -> Result<Challenge>;
```

Comment:
Hi @huoqifeng , currently we will get the challenge nonce from KBS side [here](https://github.com/confidential-containers/trustee/blob/main/kbs/src/api/src/session.rs#L78). 

And we have decoupled the code. Now the challenge will be check binding by calling Attestation Service's [reportdata/runtimedata api parameter](https://github.com/confidential-containers/trustee/blob/main/kbs/src/api/src/attestation/coco/grpc.rs#L104C9-L104C100)

So I think what is needed to be done on the verifier side is only (If SE is similiar with TDX/SNP/CCA ...)
1. implement a verifier plugin of `attester`
2. add se to https://github.com/confidential-containers/trustee/blob/main/attestation-service/attestation-service/src/bin/grpc/mod.rs#L29-L41, https://github.com/confidential-containers/trustee/blob/main/attestation-service/attestation-service/src/bin/restful/mod.rs#L54-L69, https://github.com/confidential-containers/trustee/blob/main/attestation-service/protos/attestation.proto#L5-L15
3. Document update
(I might ignore something else that could be figured out due to CI error)

Reply:
@Xynnn007 thanks for the good suggestions, there are some specific requirements for IBM SE, like:
- On kbs side, the `challenge` need include an `attestation request`.
like https://github.com/confidential-containers/trustee/blob/main/kbs/src/api/src/session.rs#L78, IBM SE has an `attestation request` need be generated by `verifier`, this request `must` be generated on verifier and pass to attester. We can use the `nonce` field but the meaning has been changed from a random str to something generated from verifier driver, alternative as I made in this draft PR, we can also use field `extra_params`, anyway, IBM SE need pass `attestation request` generated by verifier but not only random string.
- On guest side, this `attestation request` will be used to issue evidence
Like in https://github.com/confidential-containers/guest-components/blob/main/attestation-agent/kbs_protocol/src/client/rcar_client.rs#L184, we need `runtime_data` can be deserialized in `attester` driver, because IBM SE attester requires the `attestation request` passed from verifier side to generate the evidence. 

Lets talk about it this afternoon and make it clear. Thanks!

Reply:
@Xynnn007 https://github.com/ibm-s390-linux/s390-tools/tree/master/pvattest, this is a reference for c approach, rust is in-developing.